core/lib/types/src/prover_dal.rs

```diff
@@ -0,0 +1,217 @@
+//! Types exposed by the prover DAL for general-purpose use.
```

Comment:
Actually some types are used outside of the prover workspace - otherwise I wouldn't move it to the types and would've left it as is. But the point with basic_types is actually good.